lib/checks/lists/only-listitems.js

```diff
@@ -15,33 +15,39 @@ let base = {
 	liItemsWithRole: 0
 };
 
-let out = virtualNode.children.reduce((out, { actualNode }) => {
-	const tagName = actualNode.nodeName.toUpperCase();
+let out = virtualNode.children
+	.filter(node => node.nodeType !== Node.COMMENT_NODE)
+	.reduce((out, { actualNode }) => {
+		const tagName = actualNode.nodeName.toUpperCase();
 
-	if (actualNode.nodeType === 1 && dom.isVisible(actualNode, true, false)) {
-		const role = (actualNode.getAttribute('role') || '').toLowerCase();
-		const isListItemRole = getIsListItemRole(role, tagName);
+		if (actualNode.nodeType === 1 && dom.isVisible(actualNode, true, false)) {
```

Comment:
A lot of these formatting changes were generated with `npm fmt`, they're not critical for fixing this behavior.